RELEASES.md

```diff
@@ -1,3 +1,127 @@
+Version 1.77.0 (2024-03-21)
+==========================
+
+<a id="1.77.0-Language"></a>
+
+Language
+--------
+
+- [Reveal opaque types within the defining body for exhaustiveness checking.](https://github.com/rust-lang/rust/pull/116821/)
+- [Stabilize C-string literals.](https://github.com/rust-lang/rust/pull/117472/)
+- [Stabilize THIR unsafeck.](https://github.com/rust-lang/rust/pull/117673/)
+- [Support async recursive calls (as long as they have indirection).](https://github.com/rust-lang/rust/pull/117703/)
+- [Get rid of type-driven traversal in const-eval interning.](https://github.com/rust-lang/rust/pull/119044/)
```

Comment:
Note to self: T-lang is reconsidering this one: https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/should.20we.20revert.20the.20removal.20of.20type-driven.20traversal.20.2312161